### ISSUE TITLE
OpenWeatherMap not as nice as yr.no

### ISSUE THREAD

**Issue by frapell** [kotelnik/plasma-applet-weather-widget#11]
- When adding cities in owm, it will show the city id within owm, instead of the city name, as yr.no does.
- The data shown in the widget, looks way nicer for yr.no than owm (Adding screenshots here)
- It would be nice to add something like a blue line showing the current time among the timeline.

yr.no: http://picpaste.com/pics/weather_yrno.1451914961.png
owm: http://picpaste.com/pics/weather_owm-rMd0JYmR.1451915177.png

**Comment by frapell** [comment]
Also, time in widget graph should be local time, not UTC

**Comment by kotelnik** [comment]
Thanks a lot for all these bug reports. I'll address them as soon as possible.

The city id is ugly, please rename it (as workaround) by clicking the city id in the third column in the list of places. I should also add headers to the table... Eventually I'll add a picker so this should get automated.

What font are you using? Because I've tested only with default fonts with various sizes and DPI and it always looked better (smaller literals which actually fitted in the widget).

The timeline should always start from "now" so there shouldn't be any need of blue line. But first I need to fix the UTC/local-time issue :)

**Comment by frapell** [comment]
Oh! I was clicking the left number to rename it, didn't try on the right one... it does work, maybe it needs to be a bit more explicit.

As per the fonts, I'm using the default by Kubuntu 15.10, Oxygen-Sans size 8, 9 and 10 (not sure which one is used inside widgets... I change them all but I don't see it changes in the widget)

As per the timeline starting in "now", that's correct, didn't realize that until a couple of hours went by...

As you can see from the screenshot, the time is basically unreadable, being all mashed up together.

**Comment by kotelnik** [comment]
Unfortunately I'm still unable to replicate the issue with the overlapping (too big) font in rendered meteogram. I tried changing font to Oxygen-Sans, various sizes, changing DPI and also different screen resolutions. Meteogram stayed readable, no font overlapping. (You need to restart plasma in order to apply font changes). I'm suspecting the widget has some non-updated files (like e.g. Meteogram.qml), How did you install it? Is it installed only once? Can you reinstall or ensure that it is up to date? Here is how it should display:
![ww-github](https://cloud.githubusercontent.com/assets/1174734/12130208/e9085740-b408-11e5-923c-7a0c9cdc8730.png)
The rest of your reports will be fixed of course :)

**Comment by frapell** [comment]
I installed it from the "Get new widgets" button from the list of widgets... Now, it seems weird because when I go in there now, I don't see the "Uninstall" option for the widget...

Anyway, I removed the widget, installed it, uninstalled it, and re-installed it again.

Removed the 2 cities that come by default, and added mine (Cordoba, Argentina) from owm. Still seeing it wrong :(

Maybe I need to install some font or something?

**Comment by kotelnik** [comment]
Thanks a lot for all these tests!

The missing uninstall option is a known plasma bug:
https://bugs.kde.org/show_bug.cgi?id=355011
...well, you can add there that "uninstall" option is therefore also missing if you like :)

You don't need to install anything. I have a suspicion what could cause the different font size... In the widget code I am setting font size with pixels but PlasmaComponent.Label (which I use for showing texts) is already setting pointSize and maybe in Qt 5.4 (kubuntu default) is handling it differently. In Qt 5.5 pixelSize is preferred. I'll post a new version soon and let you know.

**Comment by kotelnik** [comment]
I tried to blindly address the issue :) Can you please test it before I publish a new version?

After git clone you can build it according README here on github. Or simpler way: call ./create-plasmoid-package.sh to prepare .plasmoid file (it is basically a zip archive with .plasmoid extension) - it will be generated in build directory. This file you can then drop on desktop to install it.

**Comment by gbcox** [comment]
I've never had much luck using owm - their search function is difficult to use and when you do get results they are often inaccurate.  Have you considered including wunderground.com?  yr.no appears to be a good service... is also fairly easy to search on and the results I have found to be accurate.

**Comment by kotelnik** [comment]
You're right, owm is sometimes inaccurate or just not responding. And yes, I've already considered other providers. It just takes time to implement it but I'll eventually get to work on it. I was planning weather.com, yahoo, accuweather... But wunderground.com looks like a paid service. Or do they have also a weak spot for FOSS you know of? :-)

**Comment by gbcox** [comment]
The payment is optional, and basically just gets rid of the ads.  They have some really nice graphics.  They do charge for their api, over a certain amount of queries - but that is no different than any of the other services - including OWM.  Regarding OWM, I've sent them feedback regarding the inaccuracy of their data.  So far, no reply.
